### PR TITLE
[FSDP] continue if param not exist in sharded load

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109117
* __->__ #109116

If I add a param and then wrap with FSDP + load state dict, when
strict=False don't hard error here.

Differential Revision: [D49170812](https://our.internmc.facebook.com/intern/diff/D49170812/)

### CODE REVIEW

File: torch/distributed/fsdp/_state_dict_utils.py [hunk 1]
```diff
@@ -604,7 +604,12 @@ def _sharded_pre_load_state_dict_hook(
             fqn_from_global_root = f"{prefix}{FSDP_PREFIX}{fqn}"
         else:
             fqn_from_global_root = f"{prefix}{fqn}"
-        param = state_dict.pop(fqn_from_global_root)
+        try:
+            param = state_dict.pop(fqn_from_global_root)
+        except KeyError:
+            logger.warning(f"Did not find param with FQN {fqn_from_global_root}, skipping it. The weight will not be filled if you expect it to be.")
+            continue  # TODO: only don't crash for strict=False
```
Comment (fegin) [inline torch/distributed/fsdp/_state_dict_utils.py:611]: The TODO is not necessary. The error will be delayed but will be captured by nn.Module.load_state_dict.

Comment (fegin) [reply torch/distributed/fsdp/_state_dict_utils.py:611]: We can add a TODO here to adding UTs, https://github.com/pytorch/pytorch/issues/109134.

**Review by fegin — APPROVED** [review]
LGTM, please fix the linter error, avoiding using f string with logging.

**Comment by b-chu** [comment]
Would you be able to include this in 2.1.0? This is a critical fix for crashes

**Comment by b-chu** [comment]
@atalman Would you be able to add this to 2.1.1?

**Comment by rohan-varma** [comment]
@b-chu I believe that 2.1.0 is already released, so unfortunately I'm a bit late for that. 

cc @atalman @osalpekar @seemethere  - can we make sure this gets into 2.1.1?

**Comment by osalpekar** [comment]
@rohan-varma Sure, it's been added to the release-tracker: https://github.com/pytorch/pytorch/issues/110961

**Pull request closed without merging**